mongodbatlas/data_source_mongodbatlas_search_indexes.go

```diff
@@ -100,25 +100,29 @@ func flattenSearchIndexes(searchIndexes []admin.ClusterSearchIndex, projectID, c
 
 	for i := range searchIndexes {
 		searchIndexesMap[i] = map[string]any{
-			"project_id":       projectID,
-			"cluster_name":     clusterName,
-			"analyzer":         searchIndexes[i].Analyzer,
-			"collection_name":  searchIndexes[i].CollectionName,
-			"database":         searchIndexes[i].Database,
-			"index_id":         searchIndexes[i].IndexID,
-			"mappings_dynamic": searchIndexes[i].Mappings.Dynamic,
```

Comment:
why are we removing mapping_dynamics here and setting it separately below? (just thinking on behaviour changes...)

Reply:
because Dynamics is inside Mappings and Mappings doesn't exist for vector, it's the same behaviour but making sure Mappings exists